sources/packages/web/src/components/students/modals/ConfirmEditApplication.vue

```diff
@@ -18,25 +17,46 @@
 <script lang="ts">
 import ModalDialogBase from "@/components/generic/ModalDialogBase.vue";
 import { useModalDialog } from "@/composables";
-import { defineComponent } from "vue";
+import { computed, defineComponent } from "vue";
 
 export default defineComponent({
+  props: {
+    isBeforeApplicationEdit: {
+      type: Boolean,
+      required: false,
+    },
+  },
   components: {
     ModalDialogBase,
   },
-  setup() {
+  setup(props) {
     const { showDialog, resolvePromise, showModal } = useModalDialog<boolean>();
     const dialogClosed = () => {
       resolvePromise(false);
     };
     const editApplication = async () => {
       resolvePromise(true);
     };
+
+    const confirmationText = computed(() =>
+      props.isBeforeApplicationEdit ? "Edit application" : "Submit",
+    );
+    const warningText = computed(() =>
+      props.isBeforeApplicationEdit
+        ? `Any edits made to your application may require the resubmission of
+        supporting information, potentially delaying your application. Are you
+        sure you want to proceed?`
+        : `Any edits made to your application may result in a new assessment, 
+        potentially delaying your application. Are you sure you want to proceed?`,
```

Comment:
We never used the string templates unless we needed a dynamic variable in it. Please use the usual double quotes.